lib/ledger_sync/adaptors/operation.rb

```diff
@@ -46,16 +48,34 @@ def self.included(base)
                     :result,
                     :response
 
-        def initialize(adaptor:, resource:)
-          raise 'Missing adaptor' if adaptor.nil?
-          raise 'Missing resource' if resource.nil?
-
-          raise "#{resource.class.name} is not a valid resource type.  Expected #{self.class.resource_klass.name}" unless resource.is_a?(self.class.resource_klass)
+        def initialize(
+          adaptor:,
+          ledger_deserializer_class: nil,
+          ledger_serializer_class: nil,
+          resource:,
+          validation_contract: nil
+        )
+          self.class.raise_if_unexpected_class(expected: self.class.resource_klass, given: resource.class)
 
           @adaptor = adaptor
+
+          unless ledger_deserializer_class.nil?
```

Comment:
`if x.present?` seems easier to me than reading `unless x.nil?` IMO

Reply:
If it's `nil` it defaults it (hence the function call two lines below).  If it's present, then it does a type check